pkg-manager/link-bins/src/index.ts

```diff
@@ -209,8 +209,12 @@ async function linkBin (cmd: CommandInfo, binsDir: string, opts?: LinkBinOptions
   }
 
   if (opts?.preferSymlinkedExecutables && !IS_WINDOWS && cmd.nodeExecPath == null) {
-    await symlinkDir(cmd.path, externalBinPath)
-    await fixBin(cmd.path, 0o755)
+    if (existsSync(cmd.path)) {
```

Comment:
The previous solution where you have used try/catch was better.

Reply:
But when the real file does not exist, `symlinkDir` will also create a soft link file

Reply:
Hm, I see

Reply:
although, I think it might be ok to create a soft link. It might be even expected. Maybe only fixBin should be skipped if the file doesn't exist.

Reply:
get, I had reuse `try-catch` block in latest commit, and You can see [there](https://github.com/pnpm/pnpm/pull/5946/files#diff-b75fbe9b1cb27bc5f3bdf09df4f7021ce922eb5fe21b5634d66c6951a26b5ccdR428) is an additional symbolic link file here